lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs

```diff
@@ -142,6 +142,18 @@ public void CreateDirectory(Uri directoryUri)
             _commandRunner.Run("dfs -mkdir " + directoryUri);
         }
 
+        public bool IsDirectory(Uri uri)
+        {
+            var stdOut = _commandRunner.Run("dfs -count " + uri).StdOut;
```

Comment:
Is there a better way to do this? Please see here: 
https://hadoop.apache.org/docs/r2.4.1/hadoop-project-dist/hadoop-common/FileSystemShell.html#test
hdfs dfs -test -d will check to see if the path is a directory.

Reply:
I had tried to use this initially but it does not return any value. It sets the value in $? but I am not sure how to run that command using the commandrunner

---
In reply to: [195886573](https://github.com/apache/reef/pull/1473#discussion_r195886573) [](ancestors = 195886573)

Reply:
fixed

---
In reply to: [197282264](https://github.com/apache/reef/pull/1473#discussion_r197282264) [](ancestors = 197282264,196918217,195886573)